File: LeetCode/0033/golang/lc0033_test.go
package lc0033

import "testing"

func Test_search(t *testing.T) {
	testcases := []struct {
		nums     []int
		target   int
		expected int
	}{
		{[]int{4, 5, 6, 7, 0, 1, 2}, 0, 4},
		{[]int{4, 5, 6, 7, 0, 1, 2}, 3, -1},
		{[]int{1}, 0, -1},
	}

	for i, tc := range testcases {
		actual := search(tc.nums, tc.target)
		if actual != tc.expected {
			t.Errorf("Testcase search#%02d (%v | target=%d) failed: want %d, got %d",
				i+1, tc.nums, tc.target, tc.expected, actual)
		}
	}
}

File: LeetCode/0033/python/lc0033_test.py
import unittest

from lc0033 import Solution

class Test_search(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.search([4, 5, 6, 7, 0, 1, 2], 0), 4)
        self.assertEqual(s.search([4, 5, 6, 7, 0, 1, 2], 3), -1)
        self.assertEqual(s.search([1], 0), -1)

if __name__ == "__main__":
    unittest.main()
